geomesa-utils/src/test/scala/org/locationtech/geomesa/utils/geotools/SimpleFeatureTypesTest.scala

```diff
@@ -297,6 +298,25 @@ class SimpleFeatureTypesTest extends Specification {
       sft.getDescriptor("name").getIndexCoverage() mustEqual(IndexCoverage.JOIN)
     }
 
+    "encode date attribute types" >> {
+      val sft: SimpleFeatureType = {
+        val builder = new SimpleFeatureTypeBuilder()
+        builder.setName("test")
+        builder.add("date", classOf[java.util.Date])
+        builder.add("sqlDate", classOf[java.sql.Date])
+        builder.add("sqlTimestamp", classOf[java.sql.Timestamp])
+        builder.buildFeatureType()
+      }
+      SimpleFeatureTypes.encodeDescriptor(sft, sft.getDescriptor(0)) mustEqual "date:Date"
+      SimpleFeatureTypes.encodeDescriptor(sft, sft.getDescriptor(1)) mustEqual "sqlDate:Date"
+      SimpleFeatureTypes.encodeDescriptor(sft, sft.getDescriptor(2)) mustEqual "sqlTimestamp:Timestamp"
+    }
+
+    "create schemas from sql dates" >> {
+      SimpleFeatureTypes.createType("test", "dtg:Timestamp,*geom:Point:srid=4326")
```

Comment:
how will this affect default date indexing?

Reply:
i think it will work ok - there may be some lingering bugs if someone actually did this, and i wouldn't recommend it. this change is needed in the arrow conversion code, when running against non-geomesa data stores. it creates a subtyped SFT based on the existing one, which can include timestamps.